libraries/botbuilder-core/botbuilder/core/bot_adapter.py

```diff
@@ -48,12 +50,25 @@ def use(self, middleware):
         """
         self._middleware.use(middleware)
 
-    async def run_middleware(self, context: TurnContext, callback: Callable=None):
+    async def run_middleware(self, context: TurnContext, callback: Callable[[TurnContext], Awaitable]= None):
```

Comment:
I don't understand the drift here from the C# codebase. The C# has [RunPipelineAsync](https://github.com/microsoft/botbuilder-dotnet/blob/ccf1643e3168a395f327d9eb3b6c351fa2855493/libraries/Microsoft.Bot.Builder/BotAdapter.cs#L158). 

I'm not clear why the name here is different, as it's going to cause a documentation confusion.

Reply:
This same function in JS is called runMiddleware and does functionally the same. Which one should I follow?

Reply:
I would stick with whichever codebase you've bene (mostly) modeling the Python SDK after. I have no strong preference.

Reply:
In previous versions of the Python SDK this function was named after JS. Ultimately I decided to change it given that I'm basing the code for the _most part_ in C# (class names, and structure-wise as well)